src/helper_functions.jl

```diff
@@ -22,6 +22,35 @@ function swap!(v, i::Int, j::Int)
     return nothing
 end
 
+function unstable_partition!(f, A)
```

Comment:
Can you benchmark the use of unstable_partition vs EarlyStopper? I'm up for switching because it's simpler, but performance is important.

Reply:
Benchmark added. To improve the last test cases we probably need something like `preset_voronoi` from the legacy branch.

Results from my system with 1.8.2

```
(n, s) = (400, 20)
jfa_voronoi!
  41.130 ms (0 allocations: 0 bytes)
dac_voronoi!
  3.201 ms (0 allocations: 0 bytes)
redac_voronoi!
exact_aux = DiscreteVoronoi.exact_aux
  1.085 ms (0 allocations: 0 bytes)
centre_anchor_aux = DiscreteVoronoi.centre_anchor_aux
  1.056 ms (0 allocations: 0 bytes)
redac_voronoi_es!
exact_aux_es = DiscreteVoronoi.exact_aux_es
  979.600 μs (0 allocations: 0 bytes)
centre_anchor_aux_es = DiscreteVoronoi.centre_anchor_aux_es
  946.800 μs (0 allocations: 0 bytes)
(n, s) = (400, 400)
jfa_voronoi!
  48.407 ms (0 allocations: 0 bytes)
dac_voronoi!
  254.303 ms (0 allocations: 0 bytes)
redac_voronoi!
exact_aux = DiscreteVoronoi.exact_aux
  7.985 ms (0 allocations: 0 bytes)
centre_anchor_aux = DiscreteVoronoi.centre_anchor_aux
  6.368 ms (0 allocations: 0 bytes)
redac_voronoi_es!
exact_aux_es = DiscreteVoronoi.exact_aux_es
  7.415 ms (0 allocations: 0 bytes)
centre_anchor_aux_es = DiscreteVoronoi.centre_anchor_aux_es
  5.867 ms (0 allocations: 0 bytes)
(n, s) = (400, 8000)
jfa_voronoi!
  64.304 ms (0 allocations: 0 bytes)
redac_voronoi!
exact_aux = DiscreteVoronoi.exact_aux
  206.388 ms (0 allocations: 0 bytes)
centre_anchor_aux = DiscreteVoronoi.centre_anchor_aux
  51.430 ms (0 allocations: 0 bytes)
redac_voronoi_es!
exact_aux_es = DiscreteVoronoi.exact_aux_es
  199.030 ms (0 allocations: 0 bytes)
centre_anchor_aux_es = DiscreteVoronoi.centre_anchor_aux_es
  50.399 ms (0 allocations: 0 bytes)
(n, s) = (400, 160000)
jfa_voronoi!
  49.841 ms (0 allocations: 0 bytes)
redac_voronoi!
exact_aux = DiscreteVoronoi.exact_aux
  7.645 s (0 allocations: 0 bytes)
centre_anchor_aux = DiscreteVoronoi.centre_anchor_aux
  686.620 ms (0 allocations: 0 bytes)
redac_voronoi_es!
exact_aux_es = DiscreteVoronoi.exact_aux_es
  7.466 s (0 allocations: 0 bytes)
centre_anchor_aux_es = DiscreteVoronoi.centre_anchor_aux_es
  707.790 ms (0 allocations: 0 bytes)
```

Reply:
One more remark: to replace `_findmin` with `Base.findmin`, `EarlyStopper` needs to support a `keys` method. Do you know how to implement it?